Bio/SeqIO/Interfaces.py

```diff
@@ -17,50 +17,59 @@
 
 
 class SequenceIterator:
-    """Base class for building SeqRecord iterators (DEPRECATED).
-
-    This is no longer used in the Biopython code base, instead all the parsers
-    ended up being implemented as generator functions.
+    """Base class for building SeqRecord iterators.
 
     You should write a __next__ method to return SeqRecord  objects.  You may
     wish to redefine the __init__ method as well.
     """
 
-    def __init__(self, handle, alphabet=generic_alphabet):
+    def __init__(self, source, alphabet=generic_alphabet, mode="t", fmt=None):
```

Comment:
Hmm..I couldn't find any use of `fmt` in the function body and it is not documented. Looking at how it is used in other parts of this PR, I see that it is used for passing the format string name to this base class, but I'm still not exactly sure why.

What did you have in mind for `fmt`?

Reply:
This is used in the base class in case the file mode is incorrect.
For example:
```
>>> from io import BytesIO
>>> b = BytesIO()
>>> from Bio import SeqIO
>>> SeqIO.parse(b, 'fasta')
...
Bio.StreamModeError: Fasta files must be opened in text mode.
```
The `StreamModeError` is raised in the base class; the `fmt` argument is used to have the message say "Fasta files must be opened in text mode." instead of just " File must be opened in text mode."

Reply:
Note using ``fmt`` rather than ``format`` as that is a relatively recent built in function name in Python intself.